test/dotnet.Tests/CommandTests/Project/Convert/DotnetProjectConvertTests.cs

```diff
@@ -668,6 +672,87 @@ public void ProcessingSucceeds()
                 """);
     }
 
+    [Theory, CombinatorialData]
+    public void UserSecretsId_Overridden_ViaDirective(bool hasDirectiveBuildProps)
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory();
+        File.WriteAllText(Path.Join(testInstance.Path, "Program.cs"), """
+            #:property UserSecretsId=MyIdFromDirective
+            Console.WriteLine();
+            """);
+
+        if (hasDirectiveBuildProps)
+        {
+            File.WriteAllText(Path.Join(testInstance.Path, "Directory.Build.props"), """
+                <Project>
+                  <PropertyGroup>
+                    <UserSecretsId>MyIdFromDirBuildProps</UserSecretsId>
+                  </PropertyGroup>
+                </Project>
+                """);
+        }
+
+        new DotnetCommand(Log, "project", "convert", "Program.cs")
+            .WithWorkingDirectory(testInstance.Path)
+            .Execute()
+            .Should().Pass();
+
+        File.ReadAllText(Path.Join(testInstance.Path, "Program", "Program.csproj"))
+            .Should().Be($"""
+                <Project Sdk="Microsoft.NET.Sdk">
+
+                  <PropertyGroup>
+                    <OutputType>Exe</OutputType>
+                    <TargetFramework>{ToolsetInfo.CurrentTargetFramework}</TargetFramework>
+                    <ImplicitUsings>enable</ImplicitUsings>
+                    <Nullable>enable</Nullable>
+                    <PublishAot>true</PublishAot>
+                    <UserSecretsId>MyIdFromDirective</UserSecretsId>
+                  </PropertyGroup>
+
+                </Project>
+
+                """);
+    }
+
+    [Fact]
+    public void UserSecretsId_Overridden_ViaDirectoryBuildProps()
```

Comment:
The following are corner cases, but it may be worthwhile to test them, just so we know the behavior.
- UserSecretsId is specified via `#:property` and the value is exactly the same as the one we determined implicitly.
- UserSecretsId is specified via `Directory.Build.props` and the value is exactly the same as the one we determined implicitly.